scipy/signal/_signaltools.py

```diff
@@ -1297,6 +1300,63 @@ def choose_conv_method(in1, in2, mode='full', measure=False):
     return 'direct'
 
 
+def convolve_dispatcher(in1, in2, mode='full', method='auto'):
+    """ Deduce the array namespace and if we can dispatch to the cupy namesake.
+
+    The signature must agree with that of convolve itself.
+
+    NB: Currently, raise if the namespace is CuPy and cannot dispatch to it.
+    """
+    xp = array_namespace(in1, in2)
+    if is_cupy(xp):
+        # Conditions are accurate as of CuPy 13.x
+        if method == "auto":
+            can_dispatch = in1.ndim == 1 and in2.ndim == 1
+        else:
+            can_dispatch = True
+
+        # XXX: inputs are cupy arrays and cupyx.scipy.convolve cannot be used
+        # do we want to convert to numpy/use the CPU version/convert back instead?
+        if not can_dispatch:
+            raise ValueError(
+                f"Cannot dispatch to CuPy for {in1.ndim = }, {in2.ndim = }, "
+                f"{mode = } and {method = }."
+            )
+
+    elif is_jax(xp):
+        can_dispatch = True
+    else:
+        can_dispatch = False
+
+    return xp, can_dispatch
+
+
+def dispatch_cupy(dispatcher, module_name):
+    def inner(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwds):
+            xp, can_dispatch = dispatcher(*args, **kwds)
+
+            # try delegating to a cupyx/jax namesake
+            if can_dispatch and is_cupy(xp):
+                # XXX: scipy_namespace_for(cupy) is broken
+                import importlib
+                cupyx_module = importlib.import_module(f"cupyx.scipy.{module_name}")
+                cupyx_func = getattr(cupyx_module, func.__name__)
+                return cupyx_func(*args, **kwds)
+            elif can_dispatch and is_jax(xp):
+                xps = scipy_namespace_for(xp)
```

Comment:
we should probably decide on what this variable should be called for consistency. I originally went for `spx` for the intended meaning of an `x{p}`-version of `sp` (`scipy`). I think that's a little clearer than `xps` (which is quite similar to `eps`), although I wouldn't be opposed to either `xp_sp` or `xp_scipy`.